compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -868,7 +871,45 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
         case none => true
       }
     } && !boundSym.is(TransparentImplicitMethod)
-      // FIXME: It would be nice if we could also drop type bindings, but reference counting is trickier for them.
+
+    val (termBindings, typeBindings) = bindings.partition(_.symbol.isTerm)
+
+    /** drop any referenced type symbols from the given set of type symbols */
+    val dealiasTypeBindings = new TreeMap {
+      val boundTypes = typeBindings.map(_.symbol).toSet
+
+      val dealias = new TypeMap {
+        override def apply(tp: Type) = dealiased.get(tp) match {
+          case null =>
+            val tp1 = mapOver {
+              tp match {
+                case tp: TypeRef if boundTypes.contains(tp.symbol) =>
+                  val TypeAlias(alias) = tp.info
```

Comment:
It should always be a `TypeAlias` because of the way we generate type bindings. So `boundTypes.contains(tp.symbol)` should imply `tp.info` is a `TypeAlias`.